### PR TITLE
Refactor PresentMon

### PR DESCRIPTION
**Pull request by jessicc1**
Add a CommonIncludes.hpp for commonly used headers.  Also break TraceConsumers and the definition of the PresentMonTraceConsumer out into a separate files from the PresentMonEtw.cpp source file.  Also updates PresentMonEtw.cpp to allow for more trace consumers to be added using a MultiTraceConsumer

### CODE REVIEW

File: PresentMonEtw.cpp [hunk 1]
```diff
@@ -1147,15 +157,15 @@ void PresentMonEtw(const PresentMonArgs& args)
                 // If we are reading events from ETL file set start time to match time stamp of first event
                 if (data.mArgs->mEtlFileName && data.mStartupQpcTime == 0)
                 {
-                    data.mStartupQpcTime = consumer.mTraceStartTime;
+                    data.mStartupQpcTime = pmConsumer.mTraceStartTime;
```
Comment (jenatali) [inline PresentMonEtw.cpp]: Shouldn't these be coming from procConsumer now?

File: PresentMonTraceConsumer.cpp [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+    {
+        auto result = eventInfo.GetData<uint32_t>(L"Result");
+        bool AllowBatching = SUCCEEDED(result) && result != S_PRESENT_OCCLUDED;
+        RuntimePresentStop(pEventRecord, AllowBatching);
+        break;
+    }
+    }
+}
+
+void PMTraceConsumer::OnNTProcessEvent(PEVENT_RECORD pEventRecord)
```
Comment (jenatali) [inline PresentMonTraceConsumer.cpp]: I'd go ahead and remove this from the PMTraceConsumer.

File: PresentMonTraceConsumer.hpp [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+        auto lock = scoped_lock(mProcessMutex);
+        outNewProcesses.swap(mNewProcessesFromETW);
+        outDeadProcesses.swap(mDeadProcessIds);
+    }
+
+    virtual void OnEventRecord(_In_ PEVENT_RECORD pEventRecord);
+    virtual bool ContinueProcessing() { return !g_Quit; }
+
+private:
+    void CompletePresent(std::shared_ptr<PresentEvent> p)
```
Comment (jenatali) [inline PresentMonTraceConsumer.hpp]: I'd put the private methods in the cpp. They were originally tiny helpers, and grew kind of organically into relatively large methods.

File: PresentMonEtw.cpp (hunk 1 above)
Comment (jessicc1) [reply PresentMonEtw.cpp]: Time should probably come from the owning multi-trace consumer instead.

**Review by jenatali — APPROVED** [review]
Looks good. Merging this in now.

**Pull request merged**